example-nextjs/components/CodeRunnerUI.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useRef } from 'react'
+import Editor from '@monaco-editor/react'
+import Script from 'next/script'
+import GithubButton from './GithubButton'
+import Navbar from './Navbar'
+
+interface Props {
+  initialCode: string
+  onRunCode(inputCode: string): Promise<string | void>
+  languageLabel: string
+  hideRunCode?: boolean
+  children?: React.ReactNode
+  hideOutputEditor?: boolean
+}
+
+export default function CodeRunnerUI(props: Props) {
+  const inputCodeRef = useRef(props.initialCode)
+  const [output, setOutput] = useState('')
+
+  async function runCode(code: string) {
+    console.log('running code', code)
+    const output = await props.onRunCode(inputCodeRef.current)
+    if (output) {
+      setOutput(output)
+    }
+    console.log('output', output)
```

Comment:
nit: should we remove this?

Reply:
I kept it assuming it was done for debugging at some point, but yeah, removing it would be better.